### DISCUSSION TITLE
Accessing a coupled array's du_dot_du

### DISCUSSION THREAD

**Discussion by NateAM** [idaholab/moose#25832] [Q&A General]
I'm having some trouble accessing a coupled array's _du_dot_du variable. In my header I define

`myClass.h`

    ...
    const VariableValue & _dv_dot_dv;
    ...

and in the source file I define

`myClass.C`

    InputParameters myClass::validParams( ){
        ...
        params.addRequiredCoupledVariable( "v", "my docstring" );
        ...
    }

    myClass::myClass( const InputParameters & parameters )
        : myBaseClass( parameters ),
        ...
        _v( coupledArrayValue( "v" ) ),
        _v_dot( coupledArrayDot( "v" ) ),
        _dv_dot_dv( coupledDotDu( "v" ) ),
        ...{
    }

When I run this, I get the moose error:

    *** ERROR ***
    Variable 'v' is of a different C++ type than you tried to fetch it as.

I haven't seen any class that tries to get `coupledArrayDot` so is this functionality not implemented yet? Would a workaround be to use `_du_dot_du`? What kind of limitations would that impart on my class?

**Comment by GiudGiud** [comment]
Hello

I would worry about this one `coupledDotDu( "v" )` which is likely not for Array variables (Array is not in the name)

can you see if it runs without that one?
You would then need to implement`coupledArrayDotDu( "v" )`

**Reply by NateAM** [reply]
Sorry, I was unclear! Yes, it's the `coupledDotDu` that fails for me and my subroutine works without it. I mentioned looking for `coupledArrayDot` because I was hoping to find someone else who had tried that.

I guess I should follow what was done to implement this in `Coupleable::coupledArrayVal` or `Coupleable::coupledArrayDot` and then try and add that method to my class. Does that sound about right?

**Reply by GiudGiud** [reply]
add it to Coupleable.C and then contribute it to MOOSE ;) ?

you can follow these routines. I think the regular variable coupledDotDu will also be a good inspiration

**Reply by NateAM** [reply]
> add it to Coupleable.C and then contribute it to MOOSE ;) ?

Yeah, that seems like the better option! I'll try and get this in and submit a pull request shortly. I'll need `coupledArrayDotDotDu` too so that should be included soon as well.
